### PR TITLE
add: nest+next preset + docs

### PR DESCRIPTION
**Pull request by fabriciojs**
| Issue | - |
| -----: | :-----: |
| :toolbox: Improvement | Yes |
| :trophy: Feature | Yes |
| :open_book: Documentation | Yes |
| :warning: Break Change | No |

**Description**

Adding a new preset for proposing a full Typescript mono repo with Nest for API/backend and Next for the frontend.

### CODE REVIEW

**Review by dbpolito — APPROVED** [review]

File: docs/2-Presets/1-Creating-a-Preset.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Creating a new preset
+
+`kool` has a custom task-runner like feature that allows us to automate any setup necessary for bootstrating a new project. We use this feature to enable our presets, accomplishing two objectives:
```
Comment (indiegenus) [inline docs/2-Presets/1-Creating-a-Preset.md:3]: Misspelling - `bootstrating` > `bootstrapping`

File: docs/2-Presets/1-Creating-a-Preset.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+- Keeping fast, clean and simple how to setup a new development environment for a repo and start coding.
+- Have this project with all good practices and sane defaults for running with in containers - for development and later deployment.
```
Comment (indiegenus) [inline docs/2-Presets/1-Creating-a-Preset.md:6]: Remove `with` from `...running with in containers...`

File: docs/2-Presets/1-Creating-a-Preset.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+The `config.yml` file is where we configure:
+
+- Steps for creating a new project
+- Steps for installing `kool` tailored local Docker environment to existing projects
+
+Both of the two tasks described above are accomplished via a set of declarative `steps` and `actions` on what we can call Kool Automation language. Some of the actions include:
+
+- `scripts` - running arbitrary shell script.
+- `copy` - copying files from our preset of templates folder right into the local project.
+- `merge` - merge YAML files - helpful for building `docker-compose.yml` or `kool.yml` dynamicaly.
```
Comment (indiegenus) [inline docs/2-Presets/1-Creating-a-Preset.md:23]: Misspelling - `dynamicaly` > `dynamically`

File: presets/nest+next/config.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      - copy: docker-compose.yml
+      - copy: kool.yml
+      - merge: nest+next-docker-compose.yml
+        dst: docker-compose.yml
+      - recipe: pick-db
+      - recipe: pick-cache
+      - copy: env.dist
+        dst: .env.dist
+      - copy: env.dist
+        dst: .env
```
Comment (indiegenus) [inline presets/nest+next/config.yml:27]: The `copy` and `dst` for `env.dist` are duplicated.

**Review by indiegenus — CHANGES_REQUESTED** [review]
@fabriciojs - a few (minor) changes.

File: presets/nest+next/config.yml (hunk 4 above)
Comment (fabriciojs) [reply presets/nest+next/config.yml:27]: So this one is actually correct Steve - I am copying twice, once the actual .env.dist and the second time we actually create `.env`.

The project starts then with both `.env.dist` and `.env` in place already.

**Pull request merged**